packages/fortinet_fortimail/data_stream/log/elasticsearch/ingest_pipeline/pipeline_system.yml

```
@@ -0,0 +1,171 @@
+---
+description: Pipeline for processing System logs.
+processors:
+  - set:
+      field: event.outcome
+      value: unknown
+  - set:
+      field: event.outcome
+      value: success
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'admin' && ctx.fortinet_fortimail.log.log_details?.status.toLowerCase() == 'success'
+  - set:
+      field: event.outcome
+      value: failure
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'admin' && ctx.fortinet_fortimail.log.log_details?.status.toLowerCase() == 'failure'
+  - set:
+      field: event.category
+      value: [authentication]
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'admin' && (ctx.fortinet_fortimail.log.message.toLowerCase().contains('login') || ctx.fortinet_fortimail.log.message.toLowerCase().contains('logged'))
+  - append:
+      field: event.type
+      value: end
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'admin' && (ctx.fortinet_fortimail.log.message.toLowerCase().contains('login successfully') || ctx.fortinet_fortimail.log.message.toLowerCase().contains('logged in') || ctx.fortinet_fortimail.log.message.toLowerCase().contains('login'))
```

Are these necessarily `contains` checks or are they equality checks? If they are equality checks, then it would be clearer to write something like
```suggestion
      if: ctx.fortinet_fortimail?.log?.sub_type == 'admin' && ctx.fortinet_fortimail?.log?.message != null && ['login successfully', 'logged in', 'login'].contains.(ctx.fortinet_fortimail.log.message.toLowerCase())
```
(similar elsewhere).

---

Thanks for clarifying.